backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointReadyChecker.java

```diff
@@ -75,33 +75,37 @@ public void execute() {
-                if ("ready".equals(status)) {
-                    ep.setStatus(EndpointStatus.READY);
-                }
-                if ("failed".equals(status)) {
-                    ep.setStatus(EndpointStatus.FAILED);
-                }
-                if ("deleted".equals(status)) {
-                    em.remove(ep);
-                }
-            } catch (WebApplicationException wae) {
-                // 404 means that the processor could not be found - but we were deleting an endpoint.
-                // This means is no longer there and we can finish with the delete process
-                if (wae.getResponse().getStatus() == 404 && ep.getStatus().equals(EndpointStatus.DELETING)) {
-                    em.remove(ep);
-                } else {
-                    String path = "GET " + BASE_PATH + "/{bridgeId}/processors/{processorId}";
-                    rhoseErrorMetricsRecorder.record(path, wae);
-                    Log.warn("Getting data from OB failed", wae);
-                    ep.setStatus(EndpointStatus.FAILED);
+                CamelProperties cp = em.find(CamelProperties.class, ep.getId()); // TODO Fetch in one go
+                String processorId = cp.getExtras().get("processorId");
+                try {
+                    Processor processor = bridgeApiService.getProcessorById(bridge.getId(), processorId, bridgeAuth.getToken());
+                    String status = processor.getStatus();
+                    String statusMessage = processor.getStatus_message();
+                    Log.debugf("Processor[id=%s, status=%s, status_message=%s]", processorId, status, statusMessage);
+                    if ("ready".equals(status)) {
+                        ep.setStatus(EndpointStatus.READY);
+                    }
+                    if ("failed".equals(status)) {
+                        ep.setStatus(EndpointStatus.FAILED);
+                    }
+                    if ("deleted".equals(status)) {
+                        em.remove(ep);
+                    }
+                } catch (WebApplicationException wae) {
+                    // 404 means that the processor could not be found - but we were deleting an endpoint.
+                    // This means is no longer there and we can finish with the delete process
+                    if (wae.getResponse().getStatus() == 404 && ep.getStatus().equals(EndpointStatus.DELETING)) {
+                        em.remove(ep);
+                    } else {
+                        String path = "GET " + BASE_PATH + "/{bridgeId}/processors/{processorId}";
+                        rhoseErrorMetricsRecorder.record(path, wae);
+                        Log.warn("Getting data from OB failed", wae);
+                        ep.setStatus(EndpointStatus.FAILED);
+                    }
                 }
+            } catch (Exception e) {
+                Log.error("Could not retrieve the status of a RHOSE processor", e);
```

Comment:
Would it be possible to maybe get the processor ID outside the `try {/* ... */}` block to then log it along with the error message? Maybe it is not required, it is just a suggestion :+1:

Reply:
That's a good suggestion, thanks! I've updated the code a bit.